sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/InitializePartitionProcessingContext.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+
+namespace Azure.Messaging.EventHubs.Processor
+{
+    /// <summary>
+    ///   Contains information about a partition that an <see cref="EventProcessorClient" /> will be
+    ///   processing events from.  It can also be used to specify the position within a partition
+    ///   where the associated <see cref="EventProcessorClient" /> should begin reading events in case
+    ///   it cannot find a checkpoint.
+    /// </summary>
+    ///
+    public class InitializePartitionProcessingContext
+    {
+        /// <summary>
+        ///   The context of the Event Hub partition this instance is associated with.
+        /// </summary>
+        ///
+        public PartitionContext Context { get; }
+
+        /// <summary>
+        ///   The position within a partition where the associated <see cref="EventProcessorClient" /> should
+        ///   begin reading events when no checkpoint can be found.
+        /// </summary>
+        ///
+        public EventPosition DefaultStartingPosition { internal get; set; }
```

Comment:
No need to restrict the ability to see this.   I'd go with public for both getter and setter.